stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Filters.kt

```diff
@@ -32,97 +47,59 @@ public object Filters {
 
     @JvmStatic
-    public fun greaterThanEquals(field: String, value: Any): FilterObject {
-        return FilterObject(field, FilterObject(KEY_GREATER_THAN_OR_EQUALS, value))
-    }
+    public fun lessThan(fieldName: String, value: Any): FilterObject = LessThanFilterObject(fieldName, value)
 
     @JvmStatic
-    public fun lessThan(field: String, value: Any): FilterObject {
-        return FilterObject(field, FilterObject(KEY_LESS_THAN, value))
-    }
+    public fun lessThanEquals(fieldName: String, value: Any): FilterObject = LessThanOrEqualsFilterObject(fieldName, value)
 
     @JvmStatic
-    public fun lessThanEquals(field: String, value: Any): FilterObject {
-        return FilterObject(field, FilterObject(KEY_LESS_THAN_OR_EQUALS, value))
-    }
+    public fun `in`(fieldName: String, vararg values: String): FilterObject = InFilterObject(fieldName, values.toSet())
 
     @JvmStatic
-    public fun `in`(field: String, vararg values: String): FilterObject {
-        return FilterObject(field, FilterObject(KEY_IN, values))
-    }
+    public fun `in`(fieldName: String, values: List<Any>): FilterObject = InFilterObject(fieldName, values.toSet())
 
     @JvmStatic
-    public fun `in`(field: String, values: List<*>): FilterObject {
-        return FilterObject(field, FilterObject(KEY_IN, values))
-    }
+    public fun `in`(fieldName: String, vararg values: Number): FilterObject = InFilterObject(fieldName, values.toSet())
 
     @JvmStatic
-    public fun `in`(field: String, vararg values: Number): FilterObject {
-        return FilterObject(field, FilterObject(KEY_IN, values))
-    }
+    public fun nin(fieldName: String, vararg values: String): FilterObject = NotInFilterObject(fieldName, values.toSet())
 
     @JvmStatic
-    public fun nin(field: String, vararg values: String): FilterObject {
-        return FilterObject(field, FilterObject(KEY_NOT_IN, values))
-    }
+    public fun nin(fieldName: String, values: List<Any>): FilterObject = NotInFilterObject(fieldName, values.toSet())
 
     @JvmStatic
-    public fun nin(field: String, values: List<*>): FilterObject {
-        return FilterObject(field, FilterObject(KEY_NOT_IN, values))
-    }
+    public fun nin(fieldName: String, vararg values: Number): FilterObject = NotInFilterObject(fieldName, values.toSet())
 
     @JvmStatic
-    public fun nin(field: String, vararg values: Number): FilterObject {
-        return FilterObject(field, FilterObject(KEY_NOT_IN, values))
-    }
+    public fun autocomplete(fieldName: String, value: String): FilterObject = AutocompleteFilterObject(fieldName, value)
 
     @JvmStatic
-    public fun autocomplete(field: String, value: String): FilterObject {
-        return FilterObject(field, FilterObject(KEY_AUTOCOMPLETE, value))
-    }
+    public fun distinct(memberIds: List<String>): FilterObject = DistinctFilterObject(memberIds.toSet())
```

Comment:
Is this filter _always_ used for `memberIds` specifically, and nothing else? Feels very specific.

Reply:
Yep, it is.
It is an specific filter they created for "Distinct Channels".